### PR TITLE
fix(testflight): preserve feedback fields with screenshots

### PR DESCRIPTION
**Pull request by rudrankriyam**
## Summary

- make `testflight feedback list --include-screenshots` request all 20 feedback attributes, screenshot data, and the `build`/`tester` relationship linkages
- retain the existing separate compound-resource behavior for `--include build,tester`
- cover the exact GET path/query and complete JSON response through query-builder, HTTP-client, and built command paths

## Why

PR #20 originally selected every feedback attribute modeled by the client at that time plus `screenshots`. A later model expansion added 13 documented attributes without updating that sparse fieldset, so enabling the flag caused Apple to omit them. Subsequent relationship support also exposed that sparse fieldsets must name `build` and `tester` to retain primary-resource linkages.

Apple's current public endpoint and OpenAPI continue to support this GET workflow and enumerate the complete 23-field set. The maintained AppStoreConnect Swift SDK independently defines the same public resource. There is no history or current contract evidence that the destructive field selection was intentional.

## Verification

- established RED tests showing the old eight-name fieldset
- built the CLI and issued a read-only GET with an explicit disposable app ID and file-backed authentication
  - method: `GET`
  - path: `/v1/apps/{id}/betaFeedbackScreenshotSubmissions`
  - query: exact 23-field `fields[betaFeedbackScreenshotSubmissions]` value
  - result: HTTP 200, exit 0, empty collection
  - mutation/body: none; no resource changed
- `make format`
- `make check-docs`
- `make lint`
- `ASC_BYPASS_KEYCHAIN=1 make test`
- intact pre-commit hooks: docs, format, lint, short tests

The live app had no feedback rows, so field preservation is proven deterministically by the exact HTTP and full-command response regressions; live verification proves Apple accepts the complete current query.

## Compatibility

This is additive. The command, flag, help, output modes, exit codes, and existing `--include` behavior are unchanged.

References: [Apple endpoint documentation](https://developer.apple.com/documentation/appstoreconnectapi/get-v1-apps-_id_-betafeedbackscreenshotsubmissions), [original PR #20](https://github.com/rorkai/App-Store-Connect-CLI/pull/20)

---
<a href="https://app.blacksmith.sh/rorkai/codesmith/App-Store-Connect-CLI/pr/1876"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1788627925&installation_model_id=10418&pr_number=1876&repository=rorkai%2FApp-Store-Connect-CLI&return_to=https%3A%2F%2Fgithub.com%2Frorkai%2FApp-Store-Connect-CLI%2Fpull%2F1876&signature=96ee3b4243c566d3c65034e7a0b4c535de2878e43e4644092a0ab4d5d70da2b6"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

* **Enhancements**
  * `feedback list --include-screenshots` now returns complete feedback details, including device, environment, runtime, screen, battery, and app metadata.
  * Build and tester relationships are preserved in screenshot-inclusive results.
  * JSON output consistently includes all available feedback attributes and screenshot information.

* **Documentation**
  * Added guidance covering screenshot retrieval, supported options, response details, compatibility, and edge cases.

### CODE REVIEW

**Comment by rudrankriyam** [comment]
Addressed the remaining test rework in `967564a3`:

- Added a combined CLI regression for `--include-screenshots --include build,tester` that asserts the outgoing `include=build,tester` query, preserves the complete feedback screenshot sparse fieldset, and verifies both the build and beta tester resources survive in top-level `included`.
- Replaced the screenshot payload transport stub with an `httptest.Server`, keeping request validation in the server handler and surfacing handler failures safely in the test goroutine.

The combined characterization passed before any runtime edit, so this commit intentionally changes tests only. Focused/repeated tests, full `cmdtest`, full `go test ./... -count=1`, build, docs, formatting, lint, and commit hooks are green.

**Comment by rudrankriyam** [comment]
Followed up on the exact-head CodeRabbit nit in `8ad549b8`: the payload regression now asserts representative scalar feedback values and the screenshot URL, width, height, and expiration date, in addition to preserving the complete attribute set. The focused test passed 10 consecutive runs, lint reports 0 issues, and the normal commit hook suite passed. This remains a test-only change.

File: docs/design/testflight-feedback-screenshot-fieldset.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+attributes without updating the sparse fieldset or its mock. Neither PR history
+nor review discussion indicates that discarding those attributes was intended.
+Live read-only GETs against an explicit disposable app ID returned HTTP 200 with
+and without the current screenshot fieldset; the app had no feedback rows, so
+response-field preservation remains covered deterministically by HTTP and
+command regression tests.
+
+## Behavior and compatibility
+
+JSON remains the default output; table and Markdown keep their existing
```
[the remark this answers is not in the document]
Comment (rudrankriyam) [reply docs/design/testflight-feedback-screenshot-fieldset.md]: Fixed in `c8abd41a`. The compatibility section now states the actual precedence: explicit `--output`, then `ASC_DEFAULT_OUTPUT`, then table for interactive terminals or minified JSON for pipes and CI. `make format`, `make check-docs`, `make lint`, and `ASC_BYPASS_KEYCHAIN=1 make test` pass, and the normal docs hook passed.
